questionable/private/binderror.nim

```diff
@@ -1,24 +1,43 @@
 import std/options
+import std/macros
 
-var captures {.global, compileTime.}: int
-var errorVariable {.threadvar.}: ptr ref CatchableError
+# A stack of names of error variables. Keeps track of the error variables that
+# are given to captureBindError().
+var errorVariableNames {.global, compileTime.}: seq[string]
 
-template captureBindError*(error: var ref CatchableError, expression): auto =
-  let previousErrorVariable = errorVariable
-  errorVariable = addr error
+macro captureBindError*(error: var ref CatchableError, expression): auto =
+  ## Ensures that an error is assigned to the error variable when a binding (=?)
+  ## fails inside the expression.
 
-  static: inc captures
-  let evaluated = expression
-  static: dec captures
-
-  errorVariable = previousErrorVariable
-
-  evaluated
+  # name of the error variable as a string literal
+  let errorVariableName = newLit($error)
```

Comment:
How the name generation here is working isn't quite clear to me. Looks like it's from:

```nim
  quote do:
    var error: ref CatchableError
    without captureBindError(error, `condition`):
```

The `var error` here becomes a genSym name right?

Reply:
Yes, that's correct. I can make that more explicit by adding a {.gensym.} pragma.